Replace terraform remote state with terraform>=0.9 remote backend

emalloy opened the pull request:
### Replace terraform remote state with terraform>=0.9 remote backend

This PR modifies templating around the included vpc module which now makes use of terraform 's3' remote backend and permits one to use terraform versions greater than 0.9 

This PR aims to resolve issues #32 and #30 

Other changes are:

*  `setup.py` add python encoding header, and add spaces after comment symbol
*   add state suffix to vpc/gitignore
*  add bucket target to Makefile -- creates \\`\$INFRASTRUCTURE_BUCKET` s3 bucket with bucket-versioning enabled -- if doesn't already exist
* replace main.tf with main.tf.jinja ... adding the terraform backend block for storing remote state for this module. templating because the backend resource does not support interpolation.
*  remove terraform-remote.sh in favor of ... requiring tf>0.9 and modifies Makefile  terraform instruction to replace remote with init
 

** Note: I am working on properly releasing `terraform-vpc`, see https://github.com/reactiveops/terraform-vpc/issues/31 - when this is done, I'd like to increment the references terraform-vpc module to the latest ... and include that with this PR.

emalloy committed:
remove terraform-remote.sh in favor of ... requiring tf>0.9 then modify Makefile  terraform instruction to replace remote with init

remove terraform-remote.sh.jinja in favor of +squash

realized that component install not being initialized for default/vpc +squash

emalloy committed:
replace main.tf with main.tf.jinja ... adding the terraform backend block for storing remote state for this module. templating because the backend resource does not support interpolation.

emalloy committed:
add bucket target to Makefile -- creates \\`\$INFRASTRUCTURE_BUCKET` s3 bucket with bucket-versioning enabled -- if doesn't already exist

add bucket to make all

add bucket to PHONY target, and cleanup old, removed targets

remove infrastructure-bucket stub from tasks/

emalloy committed:
add state suffix to vpc/gitignore

emalloy committed:
add python encoding header, and add spaces after comment symbol

philipcristiano commented on lib/pentagon/default/vpc/Makefile:
```diff
@@ -1,15 +1,21 @@
 .PHONY: all plan apply destroy provision ssh
```
Bucket should be added as a `PHONY` target to avoid any file shenanigans by `make`.

philipcristiano commented on lib/pentagon/components/vpc/files/Makefile:
```diff
@@ -1,19 +0,0 @@
-clean:
-	rm -f terraform.tfplan
-	rm -f terraform.tfstate
-	rm -fR .terraform/
```
Should this be removed or updated similar to the other file?

emalloy replied:
Keeping it removed --  (but open to suggestions). My thinking is that since templat-ing for the 'vpc' it's built-in to pentagon via `lib/__init__.py`, versus being installed as an explicit component.

philipcristiano reviewed (commented):
It looks like this will also remove the need for the empty stub of `/lib/pentagon/tasks/create-infrastructure-bucket.yml`. Along with maybe a pointer in https://github.com/reactiveops/pentagon/blob/15f4dfd7da9bfd3af4a3c21832e34e657a1c7d3e/bin/pentagon#L93 

Agree on review of the docs as this should simplify things.

emalloy committed:
update various pieces of documentation regarding impact of
https://github.com/reactiveops/pentagon/pull/56

update req'd version of terraform in README

philipcristiano reviewed (commented):
There is a reference to `terraform-remote.sh` in https://github.com/reactiveops/pentagon/blob/a06c9c696a4e2d88b897ae4c01359d3b4af4a082/docs/vpc.md that should be removed as well.
